contracts/neutron-interchain-txs/src/contract.rs

```diff
@@ -400,18 +430,47 @@ fn sudo_response(deps: DepsMut, request: RequestPacket, data: Binary) -> StdResu
         }
     }
 
+    ACKNOWLEDGEMENT_RESULTS.save(
+        deps.storage,
+        payload.connection_key,
+        &AcknowledgementResult::Ack(item_types),
+    )?;
+
     Ok(Response::default())
 }
 
 fn sudo_timeout(deps: DepsMut, _env: Env, request: RequestPacket) -> StdResult<Response> {
     deps.api
         .debug(format!("WASMDEBUG: sudo timeout request: {:?}", request).as_str());
+
+    let seq_id = request
+        .sequence
+        .ok_or_else(|| StdError::generic_err("sequence not found"))?;
+    let payload = read_sudo_payload(deps.storage, seq_id)?;
+
+    ACKNOWLEDGEMENT_RESULTS.save(
+        deps.storage,
+        payload.connection_key,
+        &AcknowledgementResult::Timeout(payload.message),
+    )?;
+
     Ok(Response::default())
 }
 
-fn sudo_error(deps: Deps, _request: RequestPacket, details: String) -> StdResult<Response> {
+fn sudo_error(deps: DepsMut, request: RequestPacket, details: String) -> StdResult<Response> {
     deps.api
         .debug(format!("WASMDEBUG: sudo error: {}", details).as_str());
+    let seq_id = request
+        .sequence
+        .ok_or_else(|| StdError::generic_err("sequence not found"))?;
+    let payload = read_sudo_payload(deps.storage, seq_id)?;
+
+    ACKNOWLEDGEMENT_RESULTS.save(
+        deps.storage,
+        payload.connection_key,
+        &AcknowledgementResult::Error(payload.message),
```

Comment:
I'd add the error details to the error message cuz you have the error message as simple `message` which doesn't allow you to test concrete error cases